go-controller/pkg/ovn/controller/admin_network_policy/repair.go

```diff
@@ -37,18 +35,6 @@ func (c *Controller) repairAdminNetworkPolicies() error {
 	existingANPs := map[string]*anpapi.AdminNetworkPolicy{}
 	for _, anp := range anps {
 		existingANPs[anp.Name] = anp
-		// let's populate the anpPriorityMap cache and add the correct ANP at the right priority
-		if anp.Spec.Priority > ovnkSupportedPriorityUpperBound {
```

Comment:
hm... this particular change should have been done in the other commit of this PR, right?

Reply:
I can see why this confusion came from. So let me try to explain:

Before this PR:
1. We create ANP1 at prio5-> this gets created and stored in cache
2. ANP2 is created at prio5 -> we would look at the cache and say "oops I see another ANP at same prio" I won't "load" it or "create the OVN DB plumging for this" -> but cache contains ANP1 here
3. on upgrades we restart the controller so we loose the cache memory on this situation we'd need to on repair/startup ensure that we have loaded the older configured ANP1 in cache again (not ANP2) ==> In repair this is why we needed this full block

Now this behaviour is being changed by allowing the creation of two ANPs at the same priority

After this PR:
1. We create ANP1 at prio5 -> this gets created and stored in cache
2. we create ANP2 at prio5 -> we also create this actually fully including OVN DB plumbing -> but now we use the cache and see ANP1 is present and using that we emit an event ; so the cache now is only used for emitting events nothing else
3. on upgrades we restart, we don't care about populating the right one anymore so this entire block is removed from repair.

Does that make sense?

Reply:
But... as I was explaining this to you I realised I don't have coverage for "deletion of ANP with that priority and ensuring no event is now emitted" I am going to add that now.